### PR TITLE
merge 'develop_v0.12' into 'release_v0.12'

### PR DESCRIPTION
**Pull request by Lironrad**
Minor fixes:

New Resource: elastigroup_aws_suspend_processes

resource/spotinst_ocean_ecs_launch_spec: added block_device_mappings

resource/spotinst_elastigroup_aws: added batch for integration_ecs

resource/ocean_aws: resolved error with auto_headroom_percentage under auto_scaler.

resource/spotinst_elastigroup_aws: resolved error with target_scaling_policy under auto_scaler.

documentation: resource/spotinst_mrscaler_aws: fixed usage example and documentation for retries.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,19 @@
 ## 1.26.0 (Unreleased)
+
+FEATURES:
+* *New Resource*: `elastigroup_aws_suspend_processes`
```
Comment (liranp) [inline CHANGELOG.md:4]:
```suggestion
* *New Resource*: `elastigroup_aws_suspension `
```

Comment (liranp) [inline CHANGELOG.md:4]:
```suggestion
* *New Resource*: `elastigroup_aws_suspension`
```

File: spotinst/ocean_aws_strategy/fields_spotinst_ocean_aws_strategy.go [hunk 2]
```diff
@@ -116,38 +116,43 @@ func Setup(fieldsMap map[commons.FieldName]*commons.GenericField) {
...
 			//Force setting -1 as default value if it's not exists in initial creation,
 			// to allow initialization of the field to 0
+			//SpotPercentage is configured as int in the API but float on Go-SDK (currently not aligning because of breaking changes effects)
+			//There for value is of type float and cast as necessary
 			value := spotinst.Float64(-1)
 			if cluster.Strategy != nil && cluster.Strategy.SpotPercentage != nil {
 				value = cluster.Strategy.SpotPercentage
 			}
-			if err := resourceData.Set(string(SpotPercentage), spotinst.Float64Value(value)); err != nil {
+			//Casting from float to int
```
Comment (liranp) [inline spotinst/ocean_aws_strategy/fields_spotinst_ocean_aws_strategy.go:135]: https://github.com/spotinst/terraform-provider-spotinst/pull/246#discussion_r499260839

File: spotinst/ocean_aws_strategy/fields_spotinst_ocean_aws_strategy.go [hunk 3]
```diff
@@ -116,38 +116,43 @@ func Setup(fieldsMap map[commons.FieldName]*commons.GenericField) {
...
 			}
 			return nil
 		},
 		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
 			clusterWrapper := resourceObject.(*commons.AWSClusterWrapper)
 			cluster := clusterWrapper.GetCluster()
-			if v := resourceData.Get(string(SpotPercentage)).(float64); v > -1 {
-				cluster.Strategy.SetSpotPercentage(spotinst.Float64(v))
+			if v := resourceData.Get(string(SpotPercentage)).(int); v > -1 {
+				//Casting from int to float
```
Comment (liranp) [inline spotinst/ocean_aws_strategy/fields_spotinst_ocean_aws_strategy.go:145]: https://github.com/spotinst/terraform-provider-spotinst/pull/246#discussion_r499260839

File: spotinst/ocean_aws_strategy/fields_spotinst_ocean_aws_strategy.go [hunk 4]
```diff
@@ -116,38 +116,43 @@ func Setup(fieldsMap map[commons.FieldName]*commons.GenericField) {
...
```
Comment (liranp) [inline spotinst/ocean_aws_strategy/fields_spotinst_ocean_aws_strategy.go:154]: https://github.com/spotinst/terraform-provider-spotinst/pull/246#discussion_r499260839

File: spotinst/provider.go [hunk 5]
```diff
@@ -64,6 +64,9 @@ func Provider() terraform.ResourceProvider {
 
 			// HealthCheck
 			string(commons.HealthCheckResourceName): resourceSpotinstHealthCheck(),
+
+			//SuspendProcesses
```
Comment (liranp) [inline spotinst/provider.go:68]:
```suggestion
			// SuspendProcesses
```

File: spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go [hunk 6]
```diff
@@ -0,0 +1,285 @@
+package spotinst
+
+import (
```
Comment (liranp) [inline spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go:3]: Please re-format the imports.

File: spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+func resourceSpotinstAWSSuspendProcessesCreate(resourceData *schema.ResourceData, meta interface{}) error {
+
+	log.Printf(string(commons.ResourceOnCreate), commons.SuspendProcessesResource.GetName())
+
+	suspendProcesses, err := commons.SuspendProcessesResource.OnCreate(resourceData, meta)
+	if err != nil {
+		return err
+	}
+
+	SuspendProcessesId, err := createSuspendProcesses(resourceData, suspendProcesses, meta.(*Client))
```
Comment (liranp) [inline spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go:106]:
```suggestion
	suspendProcessesID, err := createSuspendProcesses(resourceData, suspendProcesses, meta.(*Client))
```

File: spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+	if err != nil {
+		return err
+	}
+
+	resourceData.SetId(spotinst.StringValue(SuspendProcessesId))
```
Comment (liranp) [inline spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go:111]:
```suggestion
	resourceData.SetId(spotinst.StringValue(suspendProcessesID))
```

File: website/docs/r/elastigroup_aws_suspend_processes.html.markdown [hunk 9]
```diff
@@ -0,0 +1,40 @@
+---
+layout: "spotinst"
+page_title: "Spotinst: elastigroup_aws_suspend_processes"
```
Comment (liranp) [inline website/docs/r/elastigroup_aws_suspend_processes.html.markdown:3]:
```suggestion
page_title: "Spotinst: elastigroup_aws_suspension"
```

File: website/docs/r/elastigroup_aws_suspend_processes.html.markdown [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+subcategory: "Elastigroup"
+description: |-
+  Provides a Processes suspensions to Spotinst AWS group resource.
+---
+
+# spotinst\_elastigroup\_aws\_susoend_processes
```
Comment (liranp) [inline website/docs/r/elastigroup_aws_suspend_processes.html.markdown:9]:
```suggestion
# spotinst\_elastigroup\_aws\_suspension
```

File: website/docs/r/elastigroup_aws_suspend_processes.html.markdown [hunk 11]
```diff
@@ -0,0 +1,40 @@
+---
+layout: "spotinst"
+page_title: "Spotinst: elastigroup_aws_suspend_processes"
+subcategory: "Elastigroup"
+description: |-
+  Provides a Processes suspensions to Spotinst AWS group resource.
```
Comment (liranp) [inline website/docs/r/elastigroup_aws_suspend_processes.html.markdown:6]:
```suggestion
  Provides a process suspensions to Spotinst AWS group resources.
```

File: website/spotinst.erb [hunk 12]
```diff
@@ -21,6 +21,9 @@
                           <li>
                             <a href="/docs/providers/spotinst/r/elastigroup_aws_beanstalk.html">spotinst_elastigroup_aws_beanstalk</a>
                           </li>
+                          <li>
+                            <a href="/docs/providers/spotinst/r/elastigroup_aws_suspend_processes.html">spotinst_elastigroup_aws_suspension</a>
```
Comment (liranp) [inline website/spotinst.erb:25]: Please rename the file to match the resource name (`elastigroup_aws_suspension.html`).

File: spotinst/resource_spotinst_elastigroup_aws_suspend_processes_test.go [hunk 13]
```diff
@@ -0,0 +1,292 @@
...
+func testSuspendProcessesDestroy(s *terraform.State) error {
+	client := testAccProviderAWS.Meta().(*Client)
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != string(commons.SuspendProcessesResourceName) {
+			continue
+		}
+		input := &aws.DeleteSuspensionsInput{GroupID: spotinst.String(rs.Primary.ID)}
+		resp, err := client.elastigroup.CloudProviderAWS().DeleteSuspensions(context.Background(), input)
+		if err == nil && resp != nil {
+			return fmt.Errorf("Suspend Processes still exists")
```
Comment (liranp) [inline spotinst/resource_spotinst_elastigroup_aws_suspend_processes_test.go:29]:
```suggestion
			return fmt.Errorf("suspensions still exist")
```

File: spotinst/commons/common_elastigroup_aws_suspend_processes.go [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
+)
+
+var SuspendProcessesResource *SuspendProcessesTerraformResource
+
+type SuspendProcessesTerraformResource struct {
+	GenericResource // embedding
+}
+
+type SuspendProcessesWrapper struct {
+	//suspend processes is in the /aws/ec2/group/ route and groupID required as path param
```
Comment (liranp) [inline spotinst/commons/common_elastigroup_aws_suspend_processes.go:25]: No need to document it here. It's clear from the SDK that the group ID is a route parameter.

```suggestion

```

File: spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go (hunk 6 above)
Comment (Lironrad) [reply spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go:3]: I did not quite understand, I ran the commands

 `go mod tidy`  
`goimports -w ./spotinst`
`make fmt`

are there any other commands that need to be run? 
Do I need to re-format the imports manually?
@liranp

Comment (liranp) [reply spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go:3]: First, the `go mod tidy` command isn't related. Second, you have to delete the blank lines in the `import` block (specifically, lines 11 and 13) before formatting the imports.

**Pull request merged**